internal/config/config.go

```diff
@@ -221,6 +251,41 @@ func (c *SingleChainConfig) isValid() bool {
 	return isChainConfigValid
 }
 
+func (r *RoutingConfig) setDefaults() {
+	if r.Errors == nil || r.Latency == nil {
+		return
+	}
+
+	if r.DetectionWindow == nil {
+		r.DetectionWindow = newDuration(DefDetectionWindow)
+	}
+
+	if r.BanWindow == nil {
+		r.BanWindow = newDuration(DefBanWindow)
+	}
+}
+
+func (r *RoutingConfig) isRoutingConfigValidAndSetDefaults() bool {
```

Comment:
nit: Seems weird to me to combine setting defaults and validating in one function. Same on L310. WDYT about splitting?